ractor/src/factory/job.rs

```diff
@@ -83,8 +83,8 @@ impl BytesConvertable for JobOptions {
         } else {
             let ttl_bytes = data.split_off(8);
 
-            let submit_time = <u64 as BytesConvertable>::from_bytes(data);
-            let ttl = <u64 as BytesConvertable>::from_bytes(ttl_bytes);
+            let submit_time = u64::from_be_bytes(data.try_into().unwrap()); //Unwrap should be safe since we checked length earlier
```

Comment:
why?

Reply:
This was inconsistent on serialization / deserialization. the into_bytes() part of JobOptions uses ".to_be_bytes();" on the u64's, while this part fell back to the BytesConvertable implementation of u64, this obviously fails if the underlying BytesConvertable implementation changes (for example this switch to serde or even something more simple like length prefixing the encoded data). Alternative would have been to switch the encoding to use the BytesConvertible variant, but this means we can't make assumptions regarding the offsets during this decoding (namely it being 8 bytes long).